packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts

```diff
@@ -134,22 +158,24 @@ export class DynamicRemotePlugin implements WebpackPluginInstance {
       );
     }
 
-    if (compiler.options.output.uniqueName) {
+    if (this.libraryType === 'jsonp' && compiler.options.output.uniqueName) {
       throw new Error(
         'output.uniqueName option will be set automatically, do not set it in your webpack configuration',
       );
     }
 
-    compiler.options.output.uniqueName = containerName;
+    if(this.libraryType === 'jsonp') {
```

Comment:
Nit: space after `if` (lib-webpack is not currently linted, we will fix that later) here and elsewhere.